legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/ServiceLSPGrammarExtension.java

```diff
@@ -91,9 +108,55 @@ protected void collectCommands(SectionState sectionState, PackageableElement ele
     @Override
     public Iterable<? extends LegendExecutionResult> execute(SectionState section, String entityPath, String commandId, Map<String, String> executableArgs)
     {
-        return RUN_LEGACY_TESTS_COMMAND_ID.equals(commandId) ?
-                runLegacyServiceTest(section, entityPath) :
-                super.execute(section, entityPath, commandId, executableArgs);
+        switch (commandId)
+        {
+            case RUN_LEGACY_TESTS_COMMAND_ID:
+                return runLegacyServiceTest(section, entityPath);
+            case REGISTER_SERVICE_COMMAND_ID:
+                return List.of(registerService(section, entityPath));
+            default:
+                return super.execute(section, entityPath, commandId, executableArgs);
+
+        }
+    }
+
+    private LegendExecutionResult registerService(SectionState section, String entityPath)
+    {
+        try
+        {
+            PureModelContextData.Builder builder = PureModelContextData.newBuilder();
+            section.getDocumentState().getGlobalState().forEachDocumentState(docState -> docState.forEachSectionState(secState ->
+            {
+                ParseResult parseResult = secState.getProperty("parse");
+                if (parseResult != null)
+                {
+                    builder.addElements(parseResult.getElements());
+                }
+            }));
+
+            PureModelContextPointer pmcp = new PureModelContextPointer();
+            pmcp.serializer = new Protocol("pure", PureClientVersions.production);
+            pmcp.sdlcInfo = new AlloySDLC();
+            pmcp.sdlcInfo.baseVersion = "latest";
+            pmcp.sdlcInfo.version = "none";
+            pmcp.sdlcInfo.packageableElementPointers = Collections.singletonList(new PackageableElementPointer(PackageableElementType.SERVICE, entityPath));
+
+            String payload = PureProtocolObjectMapperFactory.getNewObjectMapper().writeValueAsString(builder.withOrigin(pmcp).build());
+
+            HttpResponse response = postEngineServer(REGISTER_SERVICE_FULLINTERACTIVE_URI, payload, HttpResponse.class);
```

Comment:
You will not get an HttpResponse back from postEngineServer. It will read the input stream and parse the result for you.